Kernel/Graphics/FramebufferDevice.cpp

```diff
@@ -40,6 +40,15 @@ ErrorOr<Memory::Region*> FramebufferDevice::mmap(Process& process, OpenFileDescr
     if (range.size() != framebuffer_length)
         return EOVERFLOW;
 
+    ArmedScopeGuard scope_guard = [&] {
+        if (m_real_framebuffer_region)
+            m_real_framebuffer_region = nullptr;
+        if (m_swapped_framebuffer_region)
+            m_swapped_framebuffer_region = nullptr;
+        // Note: Creation of m_userspace_framebuffer_region does not have to be
+        //       corrected due to it being the last fallible operation
+    };
+
     m_userspace_real_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_for_physical_range(m_framebuffer_address, framebuffer_length));
     m_real_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_for_physical_range(m_framebuffer_address, framebuffer_length));
     m_swapped_framebuffer_vmobject = TRY(Memory::AnonymousVMObject::try_create_with_size(framebuffer_length, AllocationStrategy::AllocateNow));
```

Comment:
This seems incorrect regardless of this fix, we shouldn't be throwing away the backing vmobject of the previous mmap.